File: DKImagePickerControllerDemo_OC/DKImagePickerControllerDemo_OC/Source/DKImagePickerController/View/Cell/DKAssetGroupDetailVideoCell.m
//
//  DKAssetGroupDetailVideoCell.m
//  DKImagePickerControllerDemo_OC
//
//  Created by 赵铭 on 2017/6/26.
//  Copyright © 2017年 zm. All rights reserved.
//

#import "DKAssetGroupDetailVideoCell.h"
#import "DKImageResource.h"
@interface DKAssetGroupDetailVideoCell()

@property (nonatomic, strong) UIView * videoInfoView;

@end


@implementation DKAssetGroupDetailVideoCell
+ (NSString *)cellReuseIdentifier{
    return @"DKVideoAssetIdentifier";
}

- (instancetype)initWithFrame:(CGRect)frame{
    if (self = [super initWithFrame: frame]) {
        [self.contentView addSubview:self.videoInfoView];
    }
    return self;
}


- (void)layoutSubviews{
    [super layoutSubviews];
    CGFloat height = 30;
    self.videoInfoView.frame = CGRectMake(0, self.contentView.bounds.size.height - height, self.contentView.bounds.size.width, height);
}

- (void)setAsset:(DKAsset *)asset{
    if (self.asset != asset) {
        super.asset = asset;
        UILabel * videoDurationLabel = [self.videoInfoView viewWithTag:-1];
        NSInteger minutes = asset.duration / 60;
        NSInteger tem = round(asset.duration);
        NSInteger seconds = tem % 60;
        videoDurationLabel.text = [NSString stringWithFormat:@"%ld:%02ld", minutes, seconds];
    }
}

- (void)setSelected:(BOOL)selected{
    super.selected = selected;
    if (selected) {
        self.videoInfoView.backgroundColor = [UIColor colorWithRed:20/255 green:129/255 blue:252/255 alpha:1];
    }else{
        self.videoInfoView.backgroundColor = [UIColor colorWithWhite:0.0 alpha:0.7];
    }
}

- (UIView *)videoInfoView{
    if (!_videoInfoView) {
        _videoInfoView = [[UIView alloc] initWithFrame:CGRectMake(0, 0, 100, 0)];
        UIImageView * videoImageView = [[UIImageView alloc] initWithImage:[DKImageResource videoCameraIcon]];
        [_videoInfoView addSubview:videoImageView];
        videoImageView.center = CGPointMake(videoImageView.bounds.size.width/2 + 7, _videoInfoView.bounds.size.height/2);
        videoImageView.autoresizingMask = UIViewAutoresizingFlexibleTopMargin | UIViewAutoresizingFlexibleBottomMargin;
        
        UILabel * videoDurationLabel = [UILabel new];
        videoDurationLabel.tag = -1;
        videoDurationLabel.textAlignment = NSTextAlignmentRight;
        videoDurationLabel.font = [UIFont systemFontOfSize:12];
        videoDurationLabel.textColor = [UIColor whiteColor];
        [_videoInfoView addSubview:videoDurationLabel];
        videoDurationLabel.frame = CGRectMake(0, 0, _videoInfoView.bounds.size.width - 7, _videoInfoView.bounds.size.height);
        videoDurationLabel.autoresizingMask = UIViewAutoresizingFlexibleWidth | UIViewAutoresizingFlexibleHeight;
    }
    return _videoInfoView;
}

/*
// Only override drawRect: if you perform custom drawing.
// An empty implementation adversely affects performance during animation.
- (void)drawRect:(CGRect)rect {
    // Drawing code
}
*/

@end

File: DKImagePickerController/View/Cell/DKAssetGroupDetailVideoCell.swift
//
//  DKAssetGroupDetailVideoCell.swift
//  DKImagePickerController
//
//  Created by ZhangAo on 07/12/2016.
//  Copyright © 2016 ZhangAo. All rights reserved.
//

import UIKit

class DKAssetGroupDetailVideoCell: DKAssetGroupDetailImageCell {
    
    class override func cellReuseIdentifier() -> String {
        return "DKVideoAssetIdentifier"
    }
    
    override init(frame: CGRect) {
        super.init(frame: frame)
        
        self.contentView.addSubview(videoInfoView)
    }
    
    required init?(coder aDecoder: NSCoder) {
        fatalError("init(coder:) has not been implemented")
    }
    
    override func layoutSubviews() {
        super.layoutSubviews()
        
        let height: CGFloat = 30
        self.videoInfoView.frame = CGRect(x: 0, y: self.contentView.bounds.height - height,
                                          width: self.contentView.bounds.width, height: height)
    }
    
    override weak var asset: DKAsset? {
        didSet {
            if let asset = asset {
                let videoDurationLabel = self.videoInfoView.viewWithTag(-1) as! UILabel
                let minutes: Int = Int(asset.duration!) / 60
                let seconds: Int = Int(round(asset.duration!)) % 60
                videoDurationLabel.text = String(format: "\(minutes):%02d", seconds)                
            }
        }
    }
    
    override var isSelected: Bool {
        didSet {
            if super.isSelected {
                self.videoInfoView.backgroundColor = UIColor(red: 20 / 255, green: 129 / 255, blue: 252 / 255, alpha: 1)
            } else {
                self.videoInfoView.backgroundColor = UIColor(white: 0.0, alpha: 0.7)
            }
        }
    }
    
    fileprivate lazy var videoInfoView: UIView = {
        let videoInfoView = UIView(frame: CGRect(x: 0, y: 0, width: 100, height: 0))
        
        let videoImageView = UIImageView(image: DKImageResource.videoCameraIcon())
        videoInfoView.addSubview(videoImageView)
        videoImageView.center = CGPoint(x: videoImageView.bounds.width / 2 + 7, y: videoInfoView.bounds.height / 2)
        videoImageView.autoresizingMask = [.flexibleBottomMargin, .flexibleTopMargin]
        
        let videoDurationLabel = UILabel()
        videoDurationLabel.tag = -1
        videoDurationLabel.textAlignment = .right
        videoDurationLabel.font = UIFont.systemFont(ofSize: 12)
        videoDurationLabel.textColor = UIColor.white
        videoInfoView.addSubview(videoDurationLabel)
        videoDurationLabel.frame = CGRect(x: 0, y: 0, width: videoInfoView.bounds.width - 7, height: videoInfoView.bounds.height)
        videoDurationLabel.autoresizingMask = [.flexibleWidth, .flexibleHeight]
        
        return videoInfoView
    }()
    
} /* DKAssetGroupDetailVideoCell */
